node/services/routes.ts

```diff
@@ -0,0 +1,53 @@
+import { ListInternalsResponse } from 'vtex.rewriter'
+import { flatten } from 'ramda'
+
+const STORE_SITEMAP_BUILD_FILE = '/dist/vtex.store-sitemap/build.json'
+
+export async function getUserRoutes(ctx: Context) {
+  const {
+    clients: { rewriter },
+  } = ctx
+
+  const LIST_LIMIT = 300
+  const internalRoutes = []
+  let nextCursor
+
+  do {
+    // eslint-disable-next-line no-await-in-loop
+    const response: ListInternalsResponse = await rewriter.listInternals(
+      LIST_LIMIT,
+      nextCursor
+    )
+    internalRoutes.push(...(response.routes ?? []))
+    nextCursor = response.next
+  } while (nextCursor)
+
+  const routes = internalRoutes.filter(
+    route =>
+      !route.disableSitemapEntry &&
+      !route.type.startsWith('notFound') &&
+      route.type !== 'product'
+  )
+
+  return routes.map(route => route.from)
+}
```

Comment:
```suggestion
const LIST_LIMIT = 300;

export async function getUserRoutes(ctx: Context) {
  const { clients: { rewriter } } = ctx;

  const internalRoutes = await fetchInternalRoutes(rewriter, LIST_LIMIT);
  const filteredRoutes = internalRoutes.filter(isValidRoute);
  return filteredRoutes.map(route => route.from);
}

async function fetchInternalRoutes(rewriter, limit) {
  const internalRoutes = [];
  let nextCursor;

  do {
    const response = await rewriter.listInternals(limit, nextCursor);
    internalRoutes.push(...(response.routes ?? []));
    nextCursor = response.next;
  } while (nextCursor);

  return internalRoutes;
}

function isValidRoute(route) {
  return (
    !route.disableSitemapEntry &&
    !route.type.startsWith('notFound') &&
    route.type !== 'product'
  );
}
```

Just splitted into different functions for readability

Reply:
Applied with internal functions on top to be lint-compatible